core/utils/hashset/hashset.c

```diff
@@ -0,0 +1,146 @@
+/* -*- Mode: C; tab-width: 4; c-basic-offset: 4; indent-tabs-mode: nil -*- */
+/*
+ *     Copyright 2012 Couchbase, Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ * 
+ * Modified in 2022 by Edward A. Lee (eal@berkeley.edu) so that stored items are
+ * consistently of type void*. Note that the void* value 1 is used to mark a deleted
+ * item and therefore cannot be stored in the hashset.
+ */
+
+#include "hashset/hashset.h"
+#include <assert.h>
+
+static const unsigned int prime_1 = 73;
+static const unsigned int prime_2 = 5009;
+
+hashset_t hashset_create()
+{
+    hashset_t set = calloc(1, sizeof(struct hashset_st));
+
+    if (set == NULL) {
+        return NULL;
+    }
+    set->nbits = 3;
+    set->capacity = (size_t)(1 << set->nbits);
+    set->mask = set->capacity - 1;
+    set->items = (void**)calloc(set->capacity, sizeof(void*));
+    if (set->items == NULL) {
+        hashset_destroy(set);
```

Comment:
we're calling hashset_destroy with set->items = NULL, we should put a NULL check on set->items before freeing it. Please see my suggestion there

Reply:
Note that this hashset_destroy is called only if memory allocation fails. In this case, there will be no hashset, so there is no need to check for null before freeing the hashset.